File: Application/Content/GetCourseContentData/GetCourseContentDataQuery.cs
using System;
using System.Collections.Immutable;
using System.Threading;
using System.Threading.Tasks;
using Biobrain.Application.Common.Core;
using Biobrain.Application.Content.ContentDataModels;
using Biobrain.Application.Content.Services.ContentCacheService;
using Biobrain.Application.Interfaces.DataAccess;
using JetBrains.Annotations;


namespace Biobrain.Application.Content.GetCourseContentData
{
    [PublicAPI]
    public sealed class GetCourseContentDataQuery : IQuery<GetCourseContentDataQuery.Result>
    {
        public Guid CourseId { get; init; }

        [PublicAPI]
        public record Result
        {
            public ImmutableList<ContentData.Page> Pages { get; init; }
            public ImmutableList<ContentData.Quiz> Quizzes { get; init; }
            public ImmutableList<ContentData.ContentTree> ContentForest { get; init; }
            public ImmutableList<ContentData.GlossaryTerm> GlossaryTerms { get; init; }
        }


        internal sealed class Handler : QueryHandlerBase<GetCourseContentDataQuery, Result>
        {
            private readonly IContentCacheService _contentCacheService;

            public Handler(IDb db, IContentCacheService contentCacheService) : base(db) => _contentCacheService = contentCacheService;

            public override async Task<Result> Handle(GetCourseContentDataQuery request, CancellationToken cancellationToken)
            {
	            var content = await _contentCacheService.GetLatestContentFromFile(request.CourseId, cancellationToken);

	            return new Result
	            {
		            Quizzes = content.Quizzes,
		            ContentForest = content.ContentForest,
		            GlossaryTerms = content.GlossaryTerms,
		            Pages = content.Pages
	            };
            }
        }
    }
}

File: BiobrainWebAngular/src/app/api/content/get-course-content-data.query.ts
import { ApiPath } from '../api-path.service';
import { Query } from '../common/query';

import { ContentTree, GlossaryTerm, MaterialPage, Quiz } from './content-data-models';

export class GetCourseContentDataQuery extends Query<GetCourseContentDataQuery_Result> {
  constructor(
    public courseId: string
  ) {
    super();
  }

  getUrl(apiPath: ApiPath): string {
    return `${apiPath.content()}/GetCourseContentData`;
  }
}


export interface GetCourseContentDataQuery_Result {
  readonly pages: MaterialPage[];
  readonly quizzes: Quiz[];
  readonly contentForest: ContentTree[];
  readonly glossaryTerms: GlossaryTerm[];
}
